### PR TITLE
CB-12189: Add support for WinMD and DLL combination

### PR DESCRIPTION
**Pull request by matrosov-nikita**
### Platforms affected
self

### What does this PR do?
This PR adds support for .winmd and .dll combination.

### What testing has been done on this change?
Auto-test

### Checklist
- [x] [Reported an issue](http://cordova.apache.org/contribute/issues.html) in the JIRA database
- [x] Commit message follows the format: "CB-3232: (android) Fix bug with resolving file paths", where CB-xxxx is the JIRA ID & "android" is the platform affected.
- [x] Added automated test coverage as appropriate for this change.

### CODE REVIEW

**Comment by vladimir-kotikov** [comment]
Documentation update: https://github.com/apache/cordova-docs/pull/671
Change to support `implementation` attribute in `cordova-common`: https://github.com/apache/cordova-lib/pull/513

Note - this PR has own override for `PluginInfo.getFrameworks` method to not depend on not-yet-released apache/cordova-lib#513 - this logic can be safely removed once updated `cordova-common` released

**Comment by vladimir-kotikov** [comment]
@daserge could you take a look please?

File: template/cordova/lib/PluginInfo.js [hunk 1]
```diff
@@ -115,6 +115,33 @@ function PluginInfo(dirname) {
         var configFiles = parentGetConfigFiles(platform);
         return processChanges(configFiles);
     };
+
+    this.getFrameworks = function(platform) {
```
Comment (daserge) [inline template/cordova/lib/PluginInfo.js]: nit: [outdated comment](https://github.com/apache/cordova-windows/pull/219/files#diff-93a26ea6e269125a5b151c4176a3b42bL100)

File: spec/unit/pluginHandler/windows.spec.js [hunk 2]
```diff
@@ -304,6 +309,47 @@ describe('windows project handler', function () {
                 xpath = 'Reference[@Include="dummy6"]/HintPath';
                 validateInstalledProjects('framework', frameworks[5], xpath, ['windows', 'windows10', 'phone']);
             });
+
+            it('with .winmd and .dll files', function() {
```
Comment (daserge) [inline spec/unit/pluginHandler/windows.spec.js]: Does this spec check that dll files have been actually copied? ([according to this](https://github.com/apache/cordova-windows/pull/219/files#diff-7bcd02ec7f03ffbb25512bd1f4a0d388R99))

**Comment by matrosov-nikita** [comment]
@vladimir-kotikov, @daserge, updated.

**Comment by vladimir-kotikov** [comment]
Rebased on top of master and force-pushed, will merge once tests pass

**Pull request closed without merging**